editor/plugins/gpu_particles_2d_editor_plugin.cpp

```diff
@@ -57,6 +57,31 @@ void GPUParticles2DEditorPlugin::_file_selected(const String &p_file) {
 	emission_mask->popup_centered();
 }
 
+void GPUParticles2DEditorPlugin::_selection_changed() {
+	List<Node *> selected_nodes = editor->get_editor_selection()->get_selected_node_list();
+
+	if (selected_particles.is_empty() && selected_nodes.is_empty()) {
+		return;
+	}
+
+	for (GPUParticles2D *SP : selected_particles) {
+		print_line(SP->get_name());
+		SP->set_show_visibility_rect(false);
+	}
+	selected_particles.clear();
+
+	if (!selected_nodes.is_empty()) {
+		for (Node *P : selected_nodes) {
+			print_line(P->get_name());
+			if (P->is_class("GPUParticles2D")) {
+				GPUParticles2D *selected_particle = Node::cast_to<GPUParticles2D>(P);
```

Comment:
Doing an inheritance check by string is slow, it's much better to rely on casting, which is what you do anyway later on (btw it's usually `Object::cast_to`, not `Node::cast_to`):
```suggestion
			GPUParticles2D *selected_particle = Object::cast_to<GPUParticles2D>(P);
			if (selected_particle) {
```

Reply:
done